File: frontend/extract_vocab.py
import json

# Load tokenizer.json
with open('lib/features/sigiriya_guide/heritageAR-chatbot/models/all-MiniLM-L6-v2/tokenizer.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Extract vocabulary (token -> id mapping)
vocab = data['model']['vocab']

# Create assets/data directory if it doesn't exist
import os
os.makedirs('assets/data', exist_ok=True)

# Write vocab.txt (one token per line, line number = token id)
with open('assets/data/vocab.txt', 'w', encoding='utf-8') as f:
    # Sort by token ID to ensure correct line numbers
    sorted_vocab = sorted(vocab.items(), key=lambda x: x[1])
    for token, _ in sorted_vocab:
        f.write(f'{token}\n')

print(f'Vocab extracted successfully: {len(vocab)} tokens')

File: frontend/bin/extract_vocab.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  // Read tokenizer.json
  final file = File('lib/features/sigiriya_guide/heritageAR-chatbot/models/all-MiniLM-L6-v2/tokenizer.json');
  final content = await file.readAsString();
  final data = jsonDecode(content) as Map<String, dynamic>;
  
  // Extract vocab
  final vocab = data['model']['vocab'] as Map<String, dynamic>;
  
  // Create assets/data directory
  final dataDir = Directory('assets/data');
  if (!await dataDir.exists()) {
    await dataDir.create(recursive: true);
  }
  
  // Sort by token ID and write to vocab.txt
  final sortedEntries = vocab.entries.toList()
    ..sort((a, b) => (a.value as int).compareTo(b.value as int));
  
  final outputFile = File('assets/data/vocab.txt');
  final sink = outputFile.openWrite();
  
  for (final entry in sortedEntries) {
    sink.writeln(entry.key);
  }
  
  await sink.close();
  
  print('Vocab extracted successfully: ${vocab.length} tokens');
}
